Lib/trace.py

```diff
@@ -559,8 +563,9 @@ def localtrace_trace_and_count(self, frame, why, arg):
             if self.start_time:
                 print('%.2f' % (_time() - self.start_time), end=' ')
             bname = os.path.basename(filename)
-            print("%s(%d): %s" % (bname, lineno,
-                                  linecache.getline(filename, lineno)), end='')
+            with os.fdopen(self.trace_fd, "w", closefd=False) as trace_fp:
```

Comment:
It uses the default encoding by default. Therefore it changes the behavior even if trace_fd=1, because the encoding of stdout can be different.

I afraid also that it can interfere with buffering user's output. Flush the standard streams before output.

Reply:
How could the code be made smarter, so that it doesn't alter the encoding of stdout?